python/clx/analytics/cybert.py

```diff
@@ -0,0 +1,227 @@
+    loading models, prediction, and postprocessing.
+    """
+
+    def __init__(self):
+        self._model = None
+        self._label_map = {}
+        self._vocab_lookup = {}
+        self._vocabpath = self.__get_vocab_file_path()
+        self._hashpath = self.__get_hash_table_path()
+
+    def load_model(
+        self, model_filepath, label_map_filepath, pretrained_model="bert-base-cased"
+    ):
+        """
+        Load cybert model.
+
+        :param model_filepath: Filepath of the model (.pth or .bin) to
+        be loaded
+        :type model_filepath: str
+        :param label_map_filepath: Filepath of the labels (.txt) to be
+        used
+        :type label_map_filepath: str
+        :param pretrained_model: Name of pretrained model to be loaded from
+        transformers
+        repo, default is bert-base-cased
+        :type pretrained_model: str
+
+        Examples
+        --------
+        >>> from clx.analytics.cybert import Cybert
+        >>> cyparse = Cybert()
+        >>> cyparse.load_model('/path/to/model.pth', '/path/to/labels.txt')
+        """
+        with open(self._vocabpath) as f:
+            for index, line in enumerate(f):
+                self._vocab_lookup[index] = line.split()[0]
+
+        with open(label_map_filepath) as f:
+            for index, line in enumerate(f):
+                self._label_map[index] = line.split()[0]
+
+        model_state_dict = torch.load(model_filepath)
+        self._model = BertForTokenClassification.from_pretrained(
+            pretrained_model,
+            state_dict=model_state_dict,
+            num_labels=len(self._label_map),
+        )
+        self._model.cuda()
+        self._model.eval()
+
+    def preprocess(self, raw_data_df, stride_len=116, max_seq_len=128):
+        """
+        Preprocess and tokenize data for cybert model inference.
+
+        :param raw_data_df: logs to be processed
+        :type model_filepath: cudf.Series
+        :param stride_len: Max stride length for processing, default is 116
+        :type stride_len: int
+        :param max_seq_len: Max sequence length for processing, default is 128
+        :type max_seq_len: int
+
+        Examples
+        --------
+        >>> import cudf
+        >>> from clx.analytics.cybert import Cybert
+        >>> cyparse = Cybert()
+        >>> cyparse.load_model('/path/to/model.pth', '/path/to/labels.txt')
+        >>> raw_df = cudf.Series(['Log event 1', 'Log event 2'])
+        >>> input_ids, attention_masks, meta_data = cyparse.preprocess(raw_df)
+        """
+        raw_data_df = raw_data_df.str.replace('"', "")
+        raw_data_df = raw_data_df.str.replace("\\r", " ")
+        raw_data_df = raw_data_df.str.replace("\\t", " ")
+        raw_data_df = raw_data_df.str.replace("=", "= ")
+        raw_data_df = raw_data_df.str.replace("\\n", " ")
+
+        self._byte_count = raw_data_df.str.byte_count()
+        self._max_num_chars = self._byte_count.sum()
```

Comment:
Set self variables like 
```
self._max_num_chars = self._byte_count.sum()
``` 
only if they'll likely be referenced outside of this function otherwise just 
```
max_num_chars = self._byte_count.sum()
```
will do. Let me know if this applies

Reply:
yup! made changes for all of your suggestions